spec/src/main/asciidoc/providers.asciidoc

```diff
@@ -47,6 +47,20 @@ The `ReaderInterceptor` interface is a listener for when a read occurs against t
 
 The `WriterInterceptor` interface is a listener for when a write occurs to the stream to be sent on the remote service invocation.
 
+=== ResponseExceptionMapper
+
+The `ResponseExceptionMapper` is specific to MicroProfile Rest Client.  This mapper will take a `Response` object retrieved via an invocation of a client and convert it to a `Throwable`, if applicable.  The runtime should scan all of the registered mappers, sort them ascending based on `getPriority()`, find the ones that can handle the given `Response`, and invoke them.  The first one discovered where `toThrowable` returns a non-empty `Optional` object will be thrown by the runtime.
```

Comment:
We're going to leave this behavior.  Its implementation specific if it calls all handles methods before invoking the toThrowable, but its only required to call toThrowable and handles until it finds a working pair.  We will always call these methods to determine if a response requires an exception to be thrown and not pre-check on response codes.  

In addition, we will require the runtime to register a default mapper that has priority `Integer.MAX_VALUE` that will map any >= 400 response into a `WebApplicationException` with the `Response` object applied.

Reply:
I think the text is out of sync with the code - the code just returns a Throwable, so instead of:

> The first one discovered where `toThrowable` returns a non-empty `Optional` object will be thrown by the runtime.

I think it should be something like:
The runtime will throw the first non-null Throwable returned from a mapper.